Opitmize LayerNormOp

xiaomengy opened the pull request:
Summary: Opitmize LayerNormOp

Differential Revision: D12398163

houseroad reviewed (approved):
LG

BIT-silence commented:
Some benchmark.
LayerNormGradientOp on CPU, Tensor Shape = [100, 100, 2000], axis = 1, 841.64ms -> 57.48ms.
LayerNormOp on GPU, Tensor Shape = [100, 1000, 2000], axis = 1, 1439.45ms -> 16.18ms
LauyerNormGradientOp on GPU, Tensor Shape = [100, 1000, 2000], axis = 1, 2148.43ms -> 22.27ms

bddppq commented:
triggered rocm tests: https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-clang7-rocmdeb-ubuntu16.04-trigger-test/3690/

xiaomengy committed:
Opitmize LayerNormOp (#13173)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13173

Opitmize LayerNormOp

Reviewed By: houseroad

Differential Revision: D12398163

fbshipit-source-id: 96fae6efd2b3eeb5917c9400a270f69b1115f360
